api/envoy/config/route/v3/route_components.proto

```diff
@@ -549,7 +550,33 @@ message CorsPolicy {
   core.v3.RuntimeFractionalPercent shadow_enabled = 10;
 }
 
-// [#next-free-field: 37]
+message HeaderRewriteLiteral {
+  // String match can use prefix match and present match. In case of prefix match, the matched prefix part of
+  // the header's value is substituted. In case of present match the whole value of the header is overwritten.
+  type.matcher.v3.StringMatcher string_match = 1 [(validate.rules).message = {required: true}];
+
+  // The matched part will be replaced by the value provided in substitution.
+  string substitution = 2 [(validate.rules).string = {min_len: 1}];
+}
+
+message HeaderMatchAndRewrite {
+  oneof header_rewriter_type {
+    // Describes the pattern to look for in the Header's value and what to substitute in its place
+    HeaderRewriteLiteral value_rewrite_literal = 1;
+  }
+
+  type.matcher.v3.MatchPredicate response_match_criteria = 2;
```

Comment:
Comments for all fields, thanks (there are some others in this PR that are not commented.

Reply:
Done